spicy/toolchain/src/compiler/codegen/codegen.cc

```diff
@@ -104,7 +104,7 @@ struct VisitorPassInit : public hilti::visitor::PreOrder<void, VisitorPassInit>
         }
 
         auto ns = cg->compileUnit(nu, false);
-        auto attrs = AttributeSet::add(t.attributes(), Attribute("&on-heap"));
+        auto attrs = AttributeSet({Attribute("&on-heap")});
```

Comment:
I checked whether this would ever throw away a existing, non-empty list of attributes and it seems that does indeed happen. Is that intentional?

Reply:
Yeah, I was trying to remember if there was a reason for transferring all existing attributes over from the original unit to the new struct type, and couldn't remember any. So changed it to only add the one attribute we want.